src/dune_pkg/rev_store.ml

```diff
@@ -48,13 +48,18 @@ let show { dir } (Rev rev) path =
 
 let load_or_create ~dir =
   let t = { dir } in
-  (* TODO might as well double check it's a directory *)
-  match Path.exists dir with
-  | true -> Fiber.return t
-  | false ->
-    Path.mkdir_p dir;
-    let+ () = run t [ "init"; "--bare" ] in
-    t
+  let+ () =
+    match Fpath.mkdir_p (Path.to_string dir) with
+    | Already_exists -> Fiber.return ()
```

Comment:
Am I correct in thinking that the directory was being created before the fiber was being run before? Does putting the creation in the `let+` make it have a more predictable outcome?